src/test/java/vetclinic/collections/katas/WhenBookingPetsIntoAPetHotel.java

```diff
@@ -0,0 +1,209 @@
+        assertThat(aPetHotel.getPets(),contains(tommy));
+        assertThat(aPetHotel.getPets().size(),is(equalTo(1)));
+    }
+
+    @Test
+    public void should_be_able_to_retrieve_checked_in_pets_in_alphabetical_order() throws Exception {
+
+        //GIVEN
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet shaggy = Pet.dog().named("Shaggy");
+        Pet bruno = Pet.dog().named("Bruno");
+
+        //WHEN
+        aPetHotel.checkIn(tommy);
+        aPetHotel.checkIn(shaggy);
+        aPetHotel.checkIn(bruno);
+
+        //THEN
+        assertThat(aPetHotel.getPets(),contains(bruno,shaggy,tommy));
+    }
+
+    @Test
+    public void should_be_able_to_obtain_a_booking_confirmation_when_we_check_in_a_pet() throws Exception {
+
+        //GIVEN
+        Pet tommy = Pet.dog().named("Tommy");
+
+        //WHEN
+        BookingConfirmation confirmation = aPetHotel.checkIn(tommy);
+
+        //THEN
+        assertThat(confirmation.isConfirmed(),is(true));
+    }
+
+    @Test
+    public void should_not_be_able_to_check_in_pets_beyond_hotel_capacity() throws Exception {
+        //GIVEN HOTEL CAPACITY IS 4
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet shaggy = Pet.dog().named("Shaggy");
+        Pet bruno = Pet.dog().named("Bruno");
+        Pet meow = Pet.cat().named("Meows");
+        aPetHotel.checkIn(tommy);
+        aPetHotel.checkIn(shaggy);
+        aPetHotel.checkIn(bruno);
+        aPetHotel.checkIn(meow);
+
+        //WHEN
+        Pet fido = Pet.dog().named("Fido");
+        aPetHotel.checkIn(fido);
+
+        //THEN
+        assertThat(aPetHotel.getPets(),not(hasItem(fido)));
+
+    }
+
+    @Test
+    public void should_notify_owner_that_the_hotel_is_full() throws Exception {
+        //GIVEN HOTEL CAPACITY IS 4
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet shaggy = Pet.dog().named("Shaggy");
+        Pet bruno = Pet.dog().named("Bruno");
+        Pet meow = Pet.cat().named("Meows");
+        aPetHotel.checkIn(tommy);
+        aPetHotel.checkIn(shaggy);
+        aPetHotel.checkIn(bruno);
+        aPetHotel.checkIn(meow);
+
+        //WHEN
+        Pet fido = Pet.dog().named("Fido");
+        BookingConfirmation bookingConfirmation =aPetHotel.checkIn(fido);
+
+        //THEN
+        assertThat(bookingConfirmation.isConfirmed(),is(false));
+        assertThat(bookingConfirmation.isWaiting(),is(true));
+    }
+
+
+    @Test
+    public void should_place_pets_on_a_waiting_list_when_the_hotel_is_full() throws Exception {
+        //GIVEN HOTEL CAPACITY IS 4
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet shaggy = Pet.dog().named("Shaggy");
+        Pet bruno = Pet.dog().named("Bruno");
+        Pet meow = Pet.cat().named("Meows");
+        aPetHotel.checkIn(tommy);
+        aPetHotel.checkIn(shaggy);
+        aPetHotel.checkIn(bruno);
+        aPetHotel.checkIn(meow);
+
+        //WHEN
+        Pet fido = Pet.dog().named("Fido");
+        BookingResponse response =aPetHotel.checkIn(fido);
```

Comment:
Since the response is not the focus of this particular test, you could also simply write:
```
aPetHotel.checkIn(fido);
```